hpi-swt2/lecture-portal
# Add button to delete a course in the course settings

# Resolves #200

This PR adds the feature of 
- redirect to course view if the lecture does not exist
- redirect to root view if the course does not exist
    - done for lecture, polls, questions, feedback
 - added test for redirection and whether the delete button is rendered 

Acceptance Criteria:
- A clearly labeled button exists on the settings page of  courses
- When I click the button a popup window asks me to confirm the deletion, and advises me that deleted data can not be restored
- If I affirm the deletion the course is removed from the database and all views, as well as students displays
- If a student is currently interacting with the lectures errors are handled gracefully
- If I don't affirm the deletion nothing happens


Testing steps:
- create a course
- go to the edit page
- use the delete button and accept
- the newly created course should now be deleted

---

Definition Of Done:
